### PR TITLE
Chore/refactor validator

### PR DESCRIPTION
**Pull request by FredericEspiau**
### Description of the Change

Validation service is now more maintenable, more readable, more coherent, while still being compatible with previous usages.

### CODE REVIEW

File: client/app/components/validator/email/validator.email.service.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        /**
+         * Determines if an email address is valid according to a set of rules
+         * @param {String} emailAddress The email address to validate
+         * @param {Object} [options] The options
+         * @param {boolean} [options.isOnlyLocalPart=false] True if only the local part (before the @) part of the email is submitted
+         * @return {validationResult} Result of the validation
+         */
+        validate (emailAddress, options) {
+            if (!_(emailAddress).isString()) {
+                throw new TypeError("domainName argument must be a String");
```
Comment (cbourgois) [inline client/app/components/validator/email/validator.email.service.js]: Fix the TypeError : `emailAddress argument must be a String`

File: client/app/domain/redirection/edit/domain-redirection-edit.controller.js [hunk 2]
```diff
@@ -138,19 +138,19 @@ angular.module("controllers").controller(
             if (_.isEmpty(this.redirectionTarget)) {
                 this.errors.redirectionTarget = true;
             } else if (this.redirection.isOrt) {
-                this.errors.redirectionTarget = !this.validator.isValidURL(this.redirectionTarget);
+                this.errors.redirectionTarget = !this.validator.url.isValid(this.redirectionTarget);
                 this.errors.targetLength = !this.isTargetValid();
             } else {
                 switch (this.redirection.fieldDisplayType) {
                 case "A":
                     this.errors.redirectionTarget = !this.validator.isValidIpv4(this.redirectionTarget);
```
Comment (cbourgois) [inline client/app/domain/redirection/edit/domain-redirection-edit.controller.js]: `this.validator.isValidIpv4(this.redirectionTarget)` no longer exists, should use `this.validator.ip.isValid(this.redirectionTarget, { version: 4 });`

File: client/app/domain/domain-validator.service.js [hunk 3]
```diff
@@ -303,16 +303,16 @@ angular.module("services").service(
             case "DYNHOST":
                 return this.Validator.isValidIpv4(target);
```
Comment (cbourgois) [inline client/app/domain/domain-validator.service.js]: `this.Validator.isValidIpv4(target)` no longer exists, should use `this.Validator.ip.isValid(target, { version: 4 });`

File: client/app/domain/glue/add/domain-glue-add-or-edit.controller.js [hunk 4]
```diff
@@ -43,7 +43,7 @@ angular.module("controllers").controller(
             if (!this.domain.glueRecordMultiIpSupported && model.ips.length > 1) {
                 valid = false;
             } else {
-                valid = !_.isEmpty(model.ips) && _.every(model.ips, (ip) => this.Validator.isValidIpv4(ip) || (_.get(this.domain, "glueRecordIpv6Supported", false) && this.Validator.isValidIpv6(ip)));
+                valid = !_.isEmpty(model.ips) && _.every(model.ips, (ip) => this.Validator.isValidIpv4(ip) || (_.get(this.domain, "glueRecordIpv6Supported", false) && this.Validator.ip.isValid(ip, { version: 6, cidr: "FORBIDDEN" })));
```
Comment (cbourgois) [inline client/app/domain/glue/add/domain-glue-add-or-edit.controller.js]: `this.validator.isValidIpv4(ip)` no longer exists, should use `this.validator.ip.isValid(ip, { version: 4 });`

File: client/app/domain/redirection/add/domain-redirection-add.controller.js [hunk 5]
```diff
@@ -225,7 +225,7 @@ angular.module("controllers").controller(
         ipaddrValid (ip) {
             if (this.Validator.isValidIpv4(ip)) {
```
Comment (cbourgois) [inline client/app/domain/redirection/add/domain-redirection-add.controller.js]: `this.Validator.isValidIpv4(ip)` no longer exists, should use `this.Validator.ip.isValid(ip, { version: 4 });`

File: client/app/resources/i18n/core/Messages_fr_FR.xml [hunk 6]
```diff
@@ -689,4 +688,33 @@
    <translation id="preload_tips_4" qtlid="433641">Nouvelle version de l'espace Web !</translation>
 
    <translation id="sidebar_menu_order_actions" qtlid="43558">Commander</translation>
+
+   <!-- Validator -->
+   <translation id="validator_domain_error_length">Un nom de domaine doit faire moins de {0} caractères.</translation>
+   <translation id="validator_domain_error_parts">Un nom de domaine doit être constitué d'au moins deux parties séparés par un point.</translation>
```
Comment (cbourgois) [inline client/app/resources/i18n/core/Messages_fr_FR.xml]: "deux parties sépar**ée**s"

File: client/app/resources/i18n/core/Messages_fr_FR.xml [hunk 7]
```diff
@@ -689,4 +688,33 @@
...
+   <translation id="validator_ip_error_orWithout">ou sans </translation>
+   <translation id="validator_ip_error_cidr">CIDR</translation>
+   <translation id="validator_ip_error_example">Exemple :</translation>
+   <translation id="validator_ip_error_example_ipv4">10.11.12.13</translation>
+   <translation id="validator_ip_error_example_ipv6">2001:db8:a0b:12f0::1</translation>
+   <translation id="validator_ip_error_example_cidr">/20</translation>
+
+   <translation id="validator_url_error_length">Une URL doit faire moins de {0} caractères.</translation>
+   <translation id="validator_url_error_characters">Une URL ne peut pas contenir d'espaces ou les caractères --, < ou >.</translation>
+   <translation id="validator_url_error_protocol">Une URL doit contenir l'un des protocols suivant {0}.</translation>
```
Comment (cbourgois) [inline client/app/resources/i18n/core/Messages_fr_FR.xml]: "protocol**e**s suivant**s**"

File: client/app/components/validator/email/validator.email.service.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+         * Determines if an email address is valid according to a set of rules
+         * @param {String} emailAddress The email address to validate
+         * @param {Object} [options] The options
+         * @param {boolean} [options.isOnlyLocalPart=false] True if only the local part (before the @) part of the email is submitted
+         * @return {boolean} True is the email address is valid
+         */
+        isValid (emailAddress, options) {
+            const isValid = this.validate(emailAddress, options).isValid;
+
+            return isValid;
```
Comment (cbourgois) [inline client/app/components/validator/email/validator.email.service.js]: avoid `const` creation, `return this.validate(emailAddress, options).isValid;`

Comment (FredericEspiau) [reply client/app/components/validator/email/validator.email.service.js]: I actually like it better cause I can easily add a breakpoint there, it's not annoying to read and it doesn't add complexity

Comment (cbourgois) [reply client/app/components/validator/email/validator.email.service.js]: Ok but it's the only `isValid` func that use `const + return` among 
`validatorDomain#isValid`, `validatorIP#isValid`, `validatorURL#isValid`...

**Comment by jleveugle** [comment]
Resolve conflict and merge ? :)

**Comment by jleveugle** [comment]
As i said 8 days ago, please resolve conflicts and we will be able to merge it :)

I close atm, but feel free to re-open it :)

**Pull request closed without merging**